packages/aws-cdk/lib/api/monitor/logs-monitor.ts

```diff
@@ -0,0 +1,196 @@
+import * as util from 'util';
+import * as colors from 'colors/safe';
+import { print, error } from '../../logging';
+import { flatten } from '../../util/arrays';
+import { ISDK } from '../aws-auth';
+
+// how often we should read events from CloudWatchLogs
+const SLEEP = 2_000;
+
+/**
+ * Represents a CloudWatch Log Event that will be
+ * printed to the terminal
+ */
+export interface CloudWatchLogEvent {
+  readonly message: string;
+  readonly logGroup: string;
+  readonly timestamp: Date;
+}
+
+/**
+ * Options for creating a CloudWatchLogEventMonitor
+ */
+export interface CloudWatchLogEventMonitorOptions {
+  /**
+   * The time that watch was first triggered.
+   * CloudWatch logs will be filtered using this start time
+   *
+   * @default - the current time
+   */
+  readonly hotswapTime?: Date;
+
+  /**
+   * The EventPrinter to use to print out CloudWatch events
+   *
+   * @default - a default printer will be created
+   */
+  readonly printer?: IEventPrinter;
+}
+
+export class CloudWatchLogEventMonitor {
+  /**
+   * Determines which events not to display
+   */
+  private startTime: number;
+
+  private logGroups = new Set<string>();
+  private sdk?: ISDK;
+
+  private active = false;
+
+  /**
+   * The event printer that controls printing out
+   * CloudWatchLog Events
+   */
+  private readonly printer: IEventPrinter;
+
+  constructor(options: CloudWatchLogEventMonitorOptions = {}) {
+    this.startTime = options.hotswapTime?.getTime() ?? Date.now();
+    this.printer = options.printer ?? new EventPrinter();
+
+    this.scheduleNextTick();
+  }
+
+  public addSdk(sdk: ISDK): void {
+    this.sdk = sdk;
+  }
+
+  // allows the ability to "pause" the monitor. The primary
+  // use case for this is when we are in the middle of performing a deployment
+  // and don't want to interweave all the logs together
+  public activate(): void {
+    this.active = true;
+  }
+
+  public deactivate(): void {
+    this.active = false;
+  }
+
+  /**
+   * Adds a CloudWatch log group to read log events from
+   */
+  public addLogGroups(logGroups: string[]): void {
+    logGroups.forEach(group => {
+      this.logGroups.add(group);
+    });
+  }
+
+  public resetLogGroups(): void {
+    this.logGroups = new Set();
```

Comment:
You can use [`Set.clear()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/clear), and in this way, make `this.logGroups` `readonly`.